docs/Datadog.Trace/README.md

```diff
@@ -0,0 +1,229 @@
+
+## Upgrading from 1.x to 2.0
+
+Version 2.0 of this package introduced a number of breaking changes to the API which allowed various performance improvements, added new features, and deprecated problematic ways of using the package. Most of these changes will not require any changes to your code, but some patterns are no longer supported or recommended.
+
+This section describes some of the most important breaking changes. For full details see [the release notes on GitHub](https://github.com/DataDog/dd-trace-dotnet/releases/tag/v2.0.0).
+
+### Updates for supported .NET versions
+
+Version 2.0 of _Datadog.Trace_ added support for .NET 6.0 and raised the minimum supported version of .NET Framework from .NET Framework 4.5 to .NET Framework 4.6.1. If you are currently targeting version < 4.6.1, we suggest you upgrade [in line with Microsoft's guidance](https://docs.microsoft.com/en-us/lifecycle/products/microsoft-net-framework).
+
+For full details of supported versions, see [our documentation on .NET Framework compatibility requirements](https://docs.datadoghq.com/tracing/setup_overview/compatibility_requirements/dotnet-framework) and [.NET/.NET Core compatibility requirements](https://docs.datadoghq.com/tracing/setup_overview/compatibility_requirements/dotnet-core).
+
+### Singleton Tracer instances
+
+In version 1.x, you could create new `Tracer` instances with different settings for each instance, using `new Tracer()`. From version 2.0 this constructor is marked `[Obsolete]`, and it is no longer possible to create `Tracer` instances with different settings. This was done to avoid multiple problematic patterns that were hard for users to detect.
+
+Update your code to the following:
+
+```csharp
+using Datadog.Trace;
+
+// Create your settings as before
+var settings = new TracerSettings();
+
+// var tracer = new Tracer(settings) // <- Delete this line
+Tracer.Configure(settings);          // <- Add this line
+```
+
+### TracerSettings are immutable
+
+In version 1.x the `TracerSettings` object passed to a `Tracer` instance could be subsequently modified. Depending on the changes, these may or may not have been respected. In version 2.0, an `ImmutableTracerSettings` object is created when the `Tracer` instance is configured. Subsequent changes to the `TracerSettings` instance will not be observed by `Tracer`.
+
+This change should not generally require changes to your code, but the type of `Tracer.Settings` is now an `ImmutableTracerSettings` instance, not a `TracerSettings` instance.
+
+```csharp
+using Datadog.Trace;
+
+var settings = TracerSettings.FromDefaultSources();
+settings.TraceEnabled = false;   // TracerSettings are mutable 
+
+Tracer.Configure(settings);
+
+// All properties on Settings are now read-only 
+// Tracer.Instance.Settings.TraceEnabled = false; // <- DOES NOT COMPILE
+```
+
+Also note that transport related settings have been isolated in the `ExporterSettings`. It is still encapsulated in `TracerSettings`.
```

Comment:
```suggestion
### Exporter settings

Exporter-related settings were grouped into the `TracerSettings.Exporter` property.

csharp
using Datadog.Trace;

var settings = TracerSettings.FromDefaultSources();
// settings.AgentUri = "http://localhost:8126";        // <- Delete this line
settings.Exporter.AgentUri = "http://localhost:8126";  // <- Add this line
```

(Sorry about that `csharp`. I can't nest `csharp` code block into a `suggestion` code block.)